packages/content/README.md

```diff
@@ -0,0 +1,148 @@
+# @esri/hub-content
+
+> Manage Hub content [`@esri/hub.js`](https://github.com/Esri/hub.js).
+
+### Example
+
+```bash
+npm install @esri/hub-content
+```
+
+```js
+import { getContent  } from '@esri/hub-content';
+
+// pass in a site id
+getContent("abc123")
+    .then(
+        contentModel => {
+            contentModel.title; // access content information
+        }
+    );
+```
+
+## Design
+
+Hub Content is the generic term for Hub-related items and integrated metadata. Content types include datasets, maps, documents, apps, surveys, as well as events, sites, pages, and initiatives. Content provides a common metadata model and interface for creating, viewing, and managing content - as well as content-specific attributes and associations such as map basemap, app datasets, site pages, and initiative site. 
+
+Content have common metadata, associated content or resources (e.g. owner member), and actions (update, delete). Content metadata is composed of the Portal Item + Data with content- or org-specific enrichments. For example, a Map content item has 
+- _literal metadata_ like `title`
+- _inferred metadata_ such as `coverage` (global, regional, local) derived from the geographic extent, 
+- _linked metadata_ `license` created through `licenseInfo`
+- _associated resources_ such as datasets via `operationalLayers` or `owner` (`as IHubMember`) via the `item.owner`
+
+### Methods
+
+- `getContent(id, associatedOptions, hubRequestOptions)`
+- `createContent(id, attributes, hubRequestOptions)`
+- `updateContent(id, attributes, hubRequestOptions)`
+- `deleteContent(id, hubRequestOptions)`
+
```

Comment:
Given the set of things that fall into `content`, I'm pretty sure we will want to delegate to some type-specific handlers - particularly when talking about apps/surveys/events/sites/pages - we should keep this in mind, esp re: the `create` and `update` fns, and for some, the `delete` (i.e. deleting a site should kill the teams, and domain record)

Reply:
@dbouwman yes. As I noted in the description I think it makes good sense to have type-specific methods. 

One aspect of this I'm curious about is if there is another content-facade. So something like...

```ts
let site = getContent(siteId);

deleteContent(site);

function deleteContent(content: IContentModel) {
  switch(content.type) {
    case 'dataset': 
      deleteDatasetById(content.id);
      break;
    case 'site':
      deleteSiteById(content.id);
      break;
    default:
      deleteContentById(content.id);
  }
}
```

Reply:
@ajturner yeah - a facade is a good pattern. Something we need to look into is how to do this so that we're not tightly coupling everything - hopefully dynamic imports can solve aspects of that for us

Reply:
maybe something of the form:
```
const content = ContentFactory.get(type).create(params)
```
there's a couple benefits including making things easier for testing purposes.

Reply:
A challenge with this is figuring out if/how we do any checks on the `params`. Depending on what you are creating, these things can be massive - i.e. this is exactly what Solution.js solves by using a template with parameters. However that only gets you so far... the other alternative is that the API is dumb, takes sends the params to the Portal API and passes back the response

Reply:
it's easy when you are talking about a handful of types with a handful of parameters, but I think in this case it's easier and better to be dumber.

Reply:
👍 definitely easier

Reply:
can we use JSON Schema similar to what AGO is using internally for [Webmap and Webscene specs](https://devtopia.esri.com/jere5132/schema-tooling) (and [validator](https://devtopia.esri.com/juan9976/schema-validator))?

But do this at runtime within our API methods for create/update?